packages/@react-aria/grid/src/useGrid.ts

```diff
@@ -73,6 +73,17 @@ export function useGrid<T>(props: GridProps, state: GridState<T, GridCollection<
     console.warn('An aria-label or aria-labelledby prop is required for accessibility.');
   }
 
+  // track keyboard navigation for blocking announcements in "replace" mode
+  let keyboardNavigated = useRef(false);
+  let onKeyDown = useCallback((e: KeyboardEvent) => {
+    if (/^Arrow(?:Right|Left|Up|Down)$/.test(e.key) || /^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+      keyboardNavigated.current = true;
```

Comment:
Wouldn't this cause an issue if you pressed the down arrow on the last item, for example? There won't be an update in that case, so this flag will get stuck.

Reply:
thanks good catch